src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -565,17 +565,26 @@ public class Reader
             private TimeSeriesValuesSegment.TagPointer _tagPointer;
             private LazyStringValue _tag;
             private TimeSeriesValuesSegment _currentSegment;
+            private TimeSpan? _offset;
 
-            public Reader(DocumentsOperationContext context, string documentId, string name, DateTime from, DateTime to)
+            public Reader(DocumentsOperationContext context, string documentId, string name, DateTime from, DateTime to, TimeSpan? offset)
             {
                 _context = context;
                 _documentId = documentId;
                 _name = name;
                 _table = new Table(TimeSeriesSchema, context.Transaction.InnerTransaction);
                 _tag = new LazyStringValue(null, null, 0, context);
+                _offset = offset;
 
                 _from = EnsureMillisecondsPrecision(from);
                 _to = EnsureMillisecondsPrecision(to);
+
+                if (_offset.HasValue)
+                {
+                    _from = _from.Add(_offset.Value);
+                    _to = _to.Add(_offset.Value);
+                }
+
```

Comment:
It would be easier to still require min & max to be in UTC, no?

Reply:
So what is the scope of the offset?

Reply:
The entire query

Reply:
I do not understand your comment here then.

Reply:
Instead of add flags to `GetNextRangeStart`, just always require it to be UTC
And convert before calling this code